stl/inc/chrono

```diff
@@ -5408,6 +5408,64 @@ _NODISCARD constexpr const _CharT* _Parse_chrono_format_specs(
 }
 
 namespace chrono {
+    // This functionality already exists in put_time, so why is it duplicated here?
+    // Unfortunately, put_time does not like to work with illogical dates, but the standard mandates that these are
+    // printed properly and treated specially. For example, consider operator<< for day. If day is 40, put_time asserts
+    // but the standard requires that it prints "40 is not a valid day".
+    template <class _CharT>
+    bool _Try_simple_write(basic_ostream<_CharT>& _Os, char _Type, const tm& _Time) {
+        if (_Type == 'd' || _Type == 'e') {
+            if (_Time.tm_mday < 10) {
+                _Os << (_Type == 'd' ? _CharT{'0'} : _CharT{' '});
+            }
+            _Os << _Time.tm_mday;
+            return true;
+        }
+
+        if (_Type == 'm') {
+            if (_Time.tm_mon + 1 < 10) {
+                _Os << _CharT{'0'};
+            }
+            _Os << _Time.tm_mon + 1;
+            return true;
+        }
+
+        if (_Type == 'Y') {
+            _Os << _STD format(_STATICALLY_WIDEN(_CharT, "{:04}"), _Time.tm_year + 1900);
+            return true;
+        }
+
+        if (_Type == 'y') {
+            _Os << _STD format(_STATICALLY_WIDEN(_CharT, "{:02}"), (_Time.tm_year + 1900) % 100);
+            return true;
+        }
+
+        if (_Type == 'C') {
+            _Os << _STD format(_STATICALLY_WIDEN(_CharT, "{:02}"), (_Time.tm_year + 1900) / 100);
+            return true;
+        }
+
+        if (_Type == 'F') {
+            _Try_simple_write(_Os, 'Y', _Time);
+            _Os << _CharT{'-'};
+            _Try_simple_write(_Os, 'm', _Time);
+            _Os << _CharT{'-'};
+            _Try_simple_write(_Os, 'd', _Time);
+            return true;
+        }
+
+        if (_Type == 'D') {
+            _Try_simple_write(_Os, 'm', _Time);
+            _Os << _CharT{'/'};
+            _Try_simple_write(_Os, 'd', _Time);
+            _Os << _CharT{'/'};
+            _Try_simple_write(_Os, 'y', _Time);
+            return true;
+        }

```

Comment:
Could probably make this into a big switch.

Reply:
You are stealing all my review thunder